src/m2k_impl.cpp

```diff
@@ -517,3 +520,101 @@ bool M2kImpl::getLed()
 	}
 	return on;
 }
+
+bool M2kImpl::initTemperatureCalibration()
+{
+	std::string values;
+	double temperature;
+
+	try {
+		// this method will throw an exception if temp_lut attribute does not exist
+		values = getContextAttributeValue("cal,temp_lut");
+	} catch (m2k_exception &e) {
+		return false;
+	}
+
+	auto splitValues = Utils::split(values, ",");
+
+	auto it = splitValues.begin();
+	while (it != splitValues.end()) {
+		try {
+			temperature = std::stod(*it);
+			auto parameters = std::make_shared<struct CALIBRATION_PARAMETERS>();
+			parameters->adc_offset_ch_1 = std::stoi(*(++it));
```

Comment:
you can use `std::next(it)` instead of the ++ operator https://en.cppreference.com/w/cpp/iterator/next

Reply:
Modified.